### PR TITLE
Update build-deb to make correct use of DOCKER_BUILD_PKGS

### PR DESCRIPTION
**Pull request by mlaventure**
Signed-off-by: Kenfe-Mickael Laventure mickael.laventure@gmail.com
## 

ping @tianon does this make sense or did I miss something?

### CODE REVIEW

**Comment by tianon** [comment]
Can you describe what's not working about `DOCKER_BUILD_PKGS` so I can understand what's being fixed here? :smile:

**Comment by mlaventure** [comment]
Oups!

@tianon 

`docker run ... -e DOCKER_BUILD_PKGS="debian-jessie debian-stretch" docker-dev:latest  hack/make.sh binary build-deb`

Will build debian-jessie correctly, then will display "skipping nonexistent debian-stretch" and exit.

This seems to be due to the fact that the `$builderDir` value only get appended to `$DOCKER_BUILD_PKGS` as a whole

**Comment by tianon** [comment]
Ahhh right, because `DOCKER_BUILD_PKGS` isn't a bash array already, OK :+1:

File: hack/make/build-deb [hunk 1]
```diff
@@ -41,7 +41,10 @@ set -e
 	builderDir="contrib/builder/deb/${PACKAGE_ARCH}"
 	pkgs=( $(find "${builderDir}/"*/ -type d) )
 	if [ ! -z "$DOCKER_BUILD_PKGS" ]; then
-		pkgs=( $(echo ${DOCKER_BUILD_PKGS[@]/#/$builderDir\/}) )
+		pkgs=()
+		for p in $DOCKER_BUILD_PKGS; do
+			pkgs+=( $(echo ${p/#/$builderDir\/}) )
```
Comment (tianon) [inline hack/make/build-deb]: This can just be `pkgs+=( "$builderDir/$p" )` :smile:

Comment (mlaventure) [inline hack/make/build-deb]: Ahah,  true, thanks :)

**Comment by mlaventure** [comment]
Updated to address @tianon comment, also updated `build-rpm` and `test-deb-install` while at it.

**Comment by thaJeztah** [comment]
ping @tianon ptal :heart:

File: hack/make/build-deb [hunk 2]
```diff
@@ -41,7 +41,10 @@ set -e
 	builderDir="contrib/builder/deb/${PACKAGE_ARCH}"
 	pkgs=( $(find "${builderDir}/"*/ -type d) )
 	if [ ! -z "$DOCKER_BUILD_PKGS" ]; then
-		pkgs=( $(echo ${DOCKER_BUILD_PKGS[@]/#/$builderDir\/}) )
+		pkgs=()
+		for p in $DOCKER_BUILD_PKGS; do
+			pkgs+=( "$builderDir/p" )
```
Comment (tianon) [inline hack/make/build-deb]: `/p` needs to be `/$p` :smile:

**Comment by mlaventure** [comment]
:facepalm: thanks for catching that @tianon, updated.

File: hack/make/build-rpm [hunk 3]
```diff
@@ -70,10 +70,13 @@ set -e
...
-		pkgs=( $(echo ${DOCKER_BUILD_PKGS[@]/#/$builderDir\/}) )
-	fi
-	for dir in "${pkgs[@]}"; do
+    if [ ! -z "$DOCKER_BUILD_PKGS" ]; then
+        pkgs=()
+        for p in $DOCKER_BUILD_PKGS; do
+            pkgs+=( "$builderDir/$p" )
+        done
+    fi
+    for dir in "${pkgs[@]}"; do
```
Comment (tianon) [inline hack/make/build-rpm]: Oh man, I didn't notice this one before either, sorry!  These are spaces and should be tabs...  :disappointed:

**Comment by tianon** [comment]
Hopefully last comment -- this LGTM other than whitespace :heart:

**Comment by mlaventure** [comment]
Fixed, I didn't have the tab exception in my editor configuration, I fixed that too.

**Comment by tianon** [comment]
Woot, LGTM :+1: (thanks for sticking it out)

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**